nuxeo-runtime/nuxeo-runtime-test/src/main/java/org/nuxeo/runtime/test/logging/DeprecationLoggerTest.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Salem Aouana
+ */
+
+package org.nuxeo.runtime.test.logging;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.logging.DeprecationLogger;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.LogCaptureFeature;
+import org.nuxeo.runtime.test.runner.LogFeature;
+import org.nuxeo.runtime.test.runner.RuntimeFeature;
+
+/**
+ * @since 11.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features({ RuntimeFeature.class, LogFeature.class, LogCaptureFeature.class })
+public class DeprecationLoggerTest {
+
+    protected static final String MESSAGE = "Deprecation contribution on component: oldComponent should now be contributed to extension point: newExtension";
+
+    protected static final String DEPRECATED_VERSION = "11.1";
+
+    protected static final String DEPRECATED_LOGGER = DeprecationLogger.class.getName();
+
+    @Inject
+    protected LogFeature logFeature;
+
+    @Inject
+    protected LogCaptureFeature.Result logCaptureResult;
+
+    @Test
+    @LogCaptureFeature.FilterOn(logLevel = "WARN")
+    public void shouldLogMessageAsWarning() {
+        System.setProperty(Framework.NUXEO_DEV_SYSTEM_PROP, "true");
+        logDeprecatedMessageAndVerify(true);
+    }
+
+    @Test
+    @LogCaptureFeature.FilterOn(logLevel = "WARN")
+    public void shouldNotLogMessageAsWarning() {
+        System.setProperty(Framework.NUXEO_DEV_SYSTEM_PROP, "false");
+        logDeprecatedMessageAndVerify(false);
+    }
+
+    @Test
+    @LogCaptureFeature.FilterOn(logLevel = "INFO")
+    public void shouldLogMessageAsInfo() {
+        logFeature.setLevel(DEPRECATED_LOGGER, "INFO");
+        System.setProperty(Framework.NUXEO_DEV_SYSTEM_PROP, "false");
+        logDeprecatedMessageAndVerify(true);
+    }
+
+    @Test
+    @LogCaptureFeature.FilterOn(logLevel = "INFO")
+    public void shouldNotLogMessageAsInfo() {
+        logFeature.setLevel(DEPRECATED_LOGGER, "FATAL");
```

Comment:
Could you add the `DEPRECATED_LOGGER` as an additional filter? This is because if we change the logger configuration, the test can fail.

Reply:
just to be sur i am understanding, you want to add `DEPRECATED_LOGGER` like below :
`@LogCaptureFeature.FilterOn(logLevel = "WARN", loggerName = DEPRECATED_LOGGER)`
If this is correct, then i have a little issue :) the annotation property needs a constant with code above there an error at compilation time, then if i need to do it i need to redefine the contant like : ` protected static final String DEPRECATED_LOGGER = "org.nuxeo.runtime.logging.DeprecationLogger";`  and if at any time we change the logger 
then i think we will end with a false positive test. any idea or i am missing something 

thx

Reply:
Talked in real, we can use `loggerClass` to do what we want.